pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -126,6 +127,12 @@ class ModelValidatorImpl implements ModelValidator {
             errorCollector.error(env, Messages.ModelValidatorImpl_NoEnvVars())
             valid = false
         }
+        env.variables.each { k, v ->
+            if (!SourceVersion.isIdentifier(k.key)) {
```

Comment:
A valid java identifier would be `$myvar` but I'm unsure if it would be a valid env var?

http://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.8